File: src/main/kotlin/com/wayllertt/entity/video.java
package com.wayllertt.entity;

import org.intellij.lang.annotations.Identifier;
import java.time.LocalDateTime ; // Для работы с датой/временем
import org.springframework.aot.hint.RuntimeHints;
import org.springframework.aot.hint.RuntimeHintsRegistrar;
import org.springframework.samples.petclinic.model.Entity;
import org.springframework.samples.petclinic.model.Table;
import org.springframework.samples.petclinic.model.Id;
import org.springframework.samples.petclinic.model.GeneratedValue;
import org.springframework.samples.petclinic.model.Column;

@Entity
@Table(name = "videos")
data class video(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 100)
    val title: String,

    @Column(nullable = false, unique = true)
    val url: String,

    @Column(nullable = false, length = 500)
    val description: String,

    @Column(nullable = true, length = 500)
    val cast: String,

    @Column()
    val director: String,

    @Column()
    val rating: ,
)

File: src/main/kotlin/com/wayllertt/entity/video.kt
package com.wayllertt.entity;

import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.Id
import jakarta.persistence.Column
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import java.time.LocalDateTime ; // Для работы с датой/временем

@Entity
@Table(name = "videos")
data class video(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 100)
    val title: String,

    val filename: String,

    @Column(nullable = false, unique = true)
    val url: String,

    @Column(nullable = false, length = 1000)
    val description: String,

    @Column(name = "cast_members", nullable = true, length = 500)
    val castMembers: String,

    @Column()
    val director: String,

    @Column(nullable = false)
    var views: Long = 0,

    @Column(nullable = false)
    var likes: Long = 0,

    @Column(name = "created_at", nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now()
)